src/Microsoft.SqlTools.ServiceLayer/Scripting/ScriptingObjectMatcher.cs

```diff
@@ -226,9 +226,11 @@ private static IEnumerable<ScriptingObject> MatchCriteria(string property, Func<
                 }
                 if (property.EndsWith(Wildcard, StringComparison.OrdinalIgnoreCase))
                 {
-                    matchedObjects = candidates.Where(o => propertySelector(o).StartsWith(
-                        propertySelector(o).Substring(0, propertySelector(o).Length - 1),
-                        StringComparison.OrdinalIgnoreCase));
+                    matchedObjects = candidates.Where(
+                        o => 
+                            propertySelector(o) != null && 
+                            propertySelector(o).StartsWith(propertySelector(o).Substring(0, propertySelector(o).Length - 1),
```

Comment:
>propertySelector(o).Substring(0, propertySelector(o).Length - 1), [](start = 59, length = 65)

nit: can you move this to a newline such that it's clear both parameters are being passed to .StartsWith()